### DISCUSSION TITLE
Argent X account on python

### DISCUSSION THREAD

**Discussion by otatataata** [software-mansion/starknet.py#1443] [General]
Hello,

I'm trying to interact programmatically with an Argent wallet on StarkNet using Python, specifically to call the `run_system` function of the Influence game contract. I'm encountering an error related to invalid owner signature. Here are the details:

**Contract Address:** 0x0422d33a3638dcc4c62e72e1d6942cd31eb643ef596ccac2351e0e21f6cd4bf4

**Function Call:** 
```python
name = encode_shortstring("ExtractResourceFinish")
calldata = [5, 5, 7809, 1, 1, 4329]
```

**Error Message:**
```
Transaction execution error. Data: {'execution_error': "failed txn 0x0797dd4e2c5603d6f6b0865346bd48dd2f6cc0f5d3fb517502e60c2623c128d3 reason: Transaction validation has failed: Execution failed. Failure reason: 0x617267656e742f696e76616c69642d6f776e65722d736967 ('argent/invalid-owner-sig'). Execution failed. Failure reason: 0x617267656e742f696e76616c69642d6f776e65722d736967 ('argent/invalid-owner-sig').", 'transaction_index': 0}
```

I'm using `starknet_py` and have tried both the standard `Account` class and attempts to use `ArgentAccount`. 

My questions are:
1. Is there a specific way to sign transactions for Argent wallets that differs from standard StarkNet accounts?
2. Are there any SDKs or libraries specifically designed for interacting with Argent wallets on StarkNet from Python?
3. Do I need to perform any additional steps to prepare my Argent wallet for programmatic interactions?

Any guidance or resources you could provide would be greatly appreciated. Thank you!

**Comment by franciszekjob** [comment]
Hey @otatataata ,
1. Argent allows to sign transactions both with 1 and 2 signatures. First one comes from regular signer key pair, while second signature comes from the guardian key, which is an Argent feature.
Basic Argent account (only with signer key) should normally work. To sign with both signer key and guardian key, you would have to implement custom `ArgentSigner` inheriting from `Signer`. You can have a look into [starknet-rs implementation](https://github.com/xJonathanLEI/starknet-rs/blob/master/starknet-accounts/src/factory/argent.rs), where guardian key is used.
2. As far as I know, there aren't any Python tools/SDKs specifically targeting Argent. It seems to me that all features needed to use Argent account can be implemented with starknet.py.
3. Please keep in mind that programmatic usage of Argent account with enabled 2FA may be problematic.

You're facing mentioned error, because passed signature fails the validation process. You can see how it's validated [here](https://github.com/argentlabs/argent-contracts-starknet/blob/1352198956f36fb35fa544c4e46a3507a3ec20e3/src/presets/argent_account.cairo#L976).
It would be helpful if you provide full code snippet to find the cause of you problem.


Also, this article about Argent's web wallet architecture may be very handy for you!
https://www.argent.xyz/blog/exploring-web-wallets-architecture/

Let me know if you need more help 😃
